### ISSUE TITLE
json_extract not working on android

### ISSUE THREAD

**Issue by jgtoriginal** [craftzdog/react-native-sqlite-2#62]
I'm storing json data on a table and when running locally on on iOS it all works fine.
But when trying to run on android, it crashes with the message `no such function json_extract`
This is my query.

```
SELECT DISTINCT
	json_extract(geometry, '$.coordinates') as coordinates,
	json_extract(geometry, '$.type') as type,
	json_extract(properties, '$.track_line_id') as track_line_id
FROM features
WHERE json_extract(geometry, '$.type') LIKE '%String%'
```

Is the [JSON1 extension](https://www.sqlite.org/json1.html) not working on android?

**Comment by jgtoriginal** [comment]
I've been digging a bit further and it seems that 

- [react-native-sqlite-2](https://github.com/craftzdog/react-native-sqlite-2) depends on 
- [websql](https://github.com/nolanlawson/node-websql) which depends on 
- [sqlite3](https://github.com/mapbox/node-sqlite3)

And it turns out that `sqlite3` has a build flag to specifically turn on and off the [JSON1 extension](https://www.sqlite.org/json1.html) `export NODE_SQLITE3_JSON1=no`, which is present on the [node building script](https://github.com/mapbox/node-sqlite3/blob/7cac428e8568bd3feafe79aa69227dbc037de8fc/scripts/build_against_node.sh) but doesn't seem to be present on the [webkit building script](https://github.com/mapbox/node-sqlite3/blob/7cac428e8568bd3feafe79aa69227dbc037de8fc/scripts/build_against_node_webkit.sh)

At this point it is clear to me that JSON1 is just disabled somehow on android while on iOS works fine. Thing is that I'm not sure how to enable it (for android).

Any help much appreciated @craftzdog !!

**Comment by craftzdog** [comment]
As you can see in the source here:

https://github.com/craftzdog/react-native-sqlite-2/blob/master/android/src/main/java/dog/craftz/sqlite_2/RNSqlite2Module.java#L215

It uses the Android's built-in SQLite3 library [android.database.sqlite](https://developer.android.com/reference/android/database/sqlite/package-summary) which seems that its `json_extract` is not enabled.

**Comment by jgtoriginal** [comment]
Oh!! I see!!
I actually found that there's another native sqlite library for android [sqlite-android](https://github.com/requery/sqlite-android) which implements `JSON1`.

And according to this issue on their repo -[can't able use json_each: it gives syntax error #32](https://github.com/requery/sqlite-android/issues/32)- the SQLite provided by the phone/OS doesn't have the json extension.

How hard would it be to swap to this other library on this project? Where could I get started?

Thanks so much for your response btw, @craftzdog !!

**Comment by craftzdog** [comment]
Try this and you will understand how to swap it:

https://github.com/requery/sqlite-android/blob/master/README.md#usage

**Comment by jgtoriginal** [comment]
Well I went through it, and seems that's not compatible.
At least out of the box, cause after making changes on the corresponding files, it just wouldn't assemble giving different errors.
I think we might be hitting a dead end here @craftzdog

**Comment by craftzdog** [comment]
Working branch: https://github.com/craftzdog/react-native-sqlite-2/tree/sqlite-android

**Comment by craftzdog** [comment]
Published as 3.0.0 with `beta` tag: https://www.npmjs.com/package/react-native-sqlite-2/v/3.0.0

**Comment by arun-saleth** [comment]
I am using watermelondb with sqlite.I faced same issue.Any solution ?
